packages/Ripple/src/index.js

```diff
@@ -0,0 +1,194 @@
+import Inferno   from 'inferno'
+import Component from 'inferno-component'
+import { bind }  from 'decko'
+
+export class Ripple extends Component {
+  state = {
+    ripples: []
+  }
+
+  componentDidMount() {
+    this.parent = this.ripple.parentElement
+
+    const { parent, handleMouseDown } = this
+    parent.addEventListener('mousedown', handleMouseDown)
+  }
+
+  componentWillUnmount() {
+    const { parent, handleMouseDown } = this
+    parent.removeEventListener('mousedown', handleMouseDown)
+  }
+
+  @bind
+  createListener(id) {
+    const { parent, startRemoval }  = this
+    const { ripples } = this.state
+
+    const handler = function() {
+      ripples[id].removable = true
+
+      // This will just work if the
+      // animation has ended
+      startRemoval(id)
+      parent.removeEventListener('mouseup', handler)
+    }
+
+    parent.addEventListener('mouseup', handler)
+  }
+
+  @bind
+  startRemoval(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+
+    if(!ripples[id].animationEnded || !ripples[id].removable)
+      return false
+
+    // If we can remove it we first start
+    // By decreasing the opacity to 0 and
+    // Then after the animation finishes we
+    // Remove the node
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+
+    setTimeout(e => this.removeRipple(id), opacityTiming || 300)
+  }
+
+  @bind
+  removeRipple(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+    setTimeout(() => {
+      ripples[id] = undefined
+      this.setState({ ripples })
+    }, 10000)
+  }
+
+  @bind
+  animationEnded(id) {
```

Comment:
Should be renamed to `handleAnimationEnded` for consistency, and should be moved down next to `handleMouseUp` and `handleMouseDown`